docs - update postGIS 2.5.4 docs

mkiyama opened the pull request:
Updates for Greenplum PostGIS 2.5.4 v2

--Add list of PostGIS extensions
--Add support for PostGIS TIGER geocoder, address standardizer and address rules files.
--Update install/uninstall instructions to use CREATE EXTENSION command
--Remove postgis_manager.sh script
--Remove PostGIS Raster limitation.

Link to updated HTML doc on a temporary GPDB draft doc site. 
https://docs-msk-gpdb7-pgis53-dev.cfapps.io/7-0/analytics/postGIS.html

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
           scope="external">Release 2.5.4</xref>.</p>
+      <note>If you installed Greenplum PostGIS 2.1.5, you cannot upgrade from PostGIS 2.1.5 to
+        2.5.4. You must uninstall PostGIS 2.1.5 and install PostGIS 2.5.4.</note>
+      <p>Greenplum PostGIS supports some PostGIS extensions. This table lists the Greenplum PostGIS
+        support for the PostGIS extensions. For information about the PostGIS extensions see the
```
Could make more terse by saying:
Greenplum PostGIS supports the following PostGIS extensions:
<table>
And remove the unsupported rows from the table.

mkiyama replied:
Moved the unsupported PostGIS extensions to a note after the table

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
+        2.5.4. You must uninstall PostGIS 2.1.5 and install PostGIS 2.5.4.</note>
+      <p>Greenplum PostGIS supports some PostGIS extensions. This table lists the Greenplum PostGIS
+        support for the PostGIS extensions. For information about the PostGIS extensions see the
+        PostGIS 2.5 documentation at <xref href="https://postgis.net/documentation/" format="html"
+          scope="external">https://postgis.net/documentation/</xref>.</p>
```
Could move this after the table and make the description the link.... unless we want the user to see the URL in its entirety

mkiyama replied:
Moved, updated text

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
+          <li><xref href="#topic_bgz_vcl_r1b" format="dita"/></li>
+        </ul>
+      </p>
+      <p>The Greenplum postGIS package installation adds these lines to the
```
capital P

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
+          <p dir="ltr">You use the <codeph>CREATE EXTENSION</codeph> command to enable the PostGIS
+            extension and the extensions that are used with PostGIS.</p>
+          <ol id="ol_mdh_z1k_zlb">
+            <li>To enable PostGIS and PostGIS Raster in a database, After logging into the database
```
remove capital A in After

mkiyama replied:
fixe

khannaekta commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
+              <entry>Supported. Enabled when Greenplum PostGIS is enabled.</entry>
+            </row>
+            <row>
+              <entry><codeph>postgis_raster</codeph><p>PostGIS Raster support</p></entry>
```
As part of PostGIS 2.5.4, PostGIS Raster support would be enabled along with creating the postgis extension. The extension name `postgis_raster` was only introduced in PostGIS version >=3.0.
So, we should delete this `postgis_raster` entry from the table and update the `postgis` entry stating that it installs PostGIS as well as Raster.

mkiyama replied:
Updated table. Combined postgis and postgis_raster extensions into a single row.

khannaekta commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
+                enabling the extension, reset the <codeph>search_path</codeph> and include the
+                PostGIS schema in the <codeph>seach_path</codeph> if needed.</p></li>
+            <li>If needed, you can enable the PostGIS TIGER geocoder. <p>To enable the PostGIS TIGER
+                geocoder, you must enable the <codeph>fuzzystrmatch</codeph> extension before
```
To enable `postgis_tiger_geocoder` we need both `postgis` as well as `fuzzystrmatch` extension. I think we should explicitly add it here and in the sample code too.

mkiyama replied:
updated text

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,129 @@
+CREATE EXTENSION postgis WITH SCHEMA &lt;schema_name> ; </codeblock><p>After
+                enabling the extension, reset the <codeph>search_path</codeph> and include the
+                PostGIS schema in the <codeph>seach_path</codeph> if needed.</p></li>
+            <li>If needed, you can enable the PostGIS TIGER geocoder. <p>To enable the PostGIS TIGER
+                geocoder, you must enable the <codeph>fuzzystrmatch</codeph> extension before
+                enabling the <codeph>postgis_tiger_geocoder</codeph> extension. These two sample
```
Could make more terse to avoid word repetition:
If needed, enable the PostGIS TIGER geocoder, after enabling the fuzzystrmatch extension. These two sample.....

mkiyama replied:
updated text

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -181,22 +254,47 @@ export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL</codeblock>
-            <codeph>GDAL_DATA</codeph>, <codeph>POSTGIS_ENABLE_OUTDB_RASTERS</codeph>,
+        <p dir="ltr">You use the <codeph>DROP EXTENSION</codeph> command to remove support for the
+          PostGIS extension and the extensions that are used with PostGIS.</p>
+        <p>Removing PostGIS Support from a database does not remove these PostGIS Raster environment
+          variables the <codeph>greenplum_path.sh</codeph> file: <codeph>GDAL_DATA</codeph>,
```
Support -> support
Also, I think there's a "from" missing after variables

mkiyama replied:
fixed

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -181,22 +254,47 @@ export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL</codeblock>
+          <p>Depending on the extensions you enabled for PostGIS, drop support for the extensions in
+            the database.</p>
+          <ol id="ol_ylb_cgk_zlb">
+            <li>If you enabled the address standardizer and sample rules tables, drop support for
+              those extensions. These commands drop support for those extensions from the current
+              database.<codeblock dir="ltr">DROP EXTENSION address_standardizer_data_us IF EXISTS;
```
Could avoid some word repetition but making a bit shorter? For example:
_If you enabled the address standardizer and sample rules tables, use these commands to drop support for those extensions from the current database._

mkiyama replied:
fixed

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -181,22 +254,47 @@ export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL</codeblock>
+              those extensions. These commands drop support for those extensions from the current
+              database.<codeblock dir="ltr">DROP EXTENSION address_standardizer_data_us IF EXISTS;
+DROP EXTENSION address_standardizer IF EXISTS ;</codeblock></li>
+            <li>If you enabled the TIGER geocoder and the <codeph>fuzzystrmatch</codeph> extension
+              to use the TIGER geocoder. These commands drop support for those
```
Something is missing here?

mkiyama replied:
fixed

lenapivotal commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -181,22 +254,47 @@ export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL</codeblock>
+            <li>Drop support for PostGIS and PostGIS raster. This command drops support for those
+                extensions.<codeblock>DROP EXTENSION postgis IF EXISTS ;</codeblock><p>If you
+                enabled support for PostGIS and specified a specific schema with the <codeph>CREATE
+                  EXTENSION</codeph> command, you can update the <codeph>search_path</codeph>and
```
missing space

mkiyama replied:
fixed

mkiyama commented:
Updated PR based on review comments.
Also updated the HTML output on the temp. doc site
 https://docs-msk-gpdb7-pgis53-dev.cfapps.io/7-0/analytics/postGIS.html

khannaekta commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,121 @@
+          </thead>
+          <tbody>
+            <row>
+              <entry><codeph>postgis</codeph> and <codeph>postgis_raster</codeph><p dir="ltr"
```
```suggestion
              <entry><codeph>postgis</codeph><p dir="ltr"
```
Can we remove the term "postgis_raster" to avoid confusion since that name does not exist in PostGIS 2.5.4.

mkiyama replied:
removed postgis_raster.

dyozie commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,121 @@
+          <li><xref href="#topic_bgz_vcl_r1b" format="dita"/></li>
+        </ul>
+      </p>
+      <p>The Greenplum PostGIS package installation adds these lines to the
```
It seems strange that there is no "Installing the Greenplum PostGIS Extension Package" section but there is an Uninstalling section.  I think this would be best wrapped in an "Installing" section with a one-line gppkg example, and conclude with this paragraph about the greenplum_path.sh changes.

mkiyama replied:
Added the section and other related edits.
Also updated the section on removing the PosgGIS package.

dyozie commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,121 @@
+                </p><codeblock>SHOW search_path ; -- display the current search_path
+CREATE SCHEMA &lt;schema_name> ;
+SET search_path TO &lt;schema_name> ;
+CREATE EXTENSION postgis WITH SCHEMA &lt;schema_name> ; </codeblock><p>After
```
There's a stray  character here after the semicolon, which shows up in the html as well.

mkiyama replied:
Not sure where that character keeps comes from. It is a copy and paste thing.

dyozie commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -76,53 +73,121 @@
+SET search_path TO &lt;schema_name> ;
+CREATE EXTENSION postgis WITH SCHEMA &lt;schema_name> ; </codeblock><p>After
+                enabling the extension, reset the <codeph>search_path</codeph> and include the
+                PostGIS schema in the <codeph>seach_path</codeph> if needed.</p></li>
```
Typo:  seach_path

mkiyama replied:
fixed

dyozie commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -181,22 +246,47 @@ export POSTGIS_GDAL_ENABLED_DRIVERS=DISABLE_ALL</codeblock>
-        </p>
+          when you uninstall the PostGIS extension package.</p>
+        <section id="drop_postgis_cmd">
+          <title>Using the DROP EXTENSION command</title>
```
command -> Command

mkiyama replied:
fixed

dyozie commented on gpdb-doc/dita/analytics/postGIS.xml:
```diff
@@ -349,116 +446,22 @@ SELECT AddGeometryColumn('public', 'geometries', 'geom', 0, 'LINESTRING', 2);</c
           <li>The <codeph>&lt;-></codeph> operator (<codeph><varname>geometry</varname> &lt;->
                 <varname>geometry</varname></codeph>) returns the centroid/centroid distance for
             Greenplum Database 6.</li>
+          <li>The following PostGIS functions are not currently supported because Greenplun PostGIS
```
Typo:  Greenplun

mkiyama replied:
fixed

mkiyama commented:
updated the HTML output on the temp. doc site
https://docs-msk-gpdb7-pgis53-dev.cfapps.io/7-0/analytics/postGIS.html
